modules/shapefile/src/shp-loader.js

```diff
@@ -24,5 +25,11 @@ export const SHPWorkerLoader = {
 export const SHPLoader = {
   ...SHPWorkerLoader,
   parse: async (arrayBuffer, options) => parseSHP(arrayBuffer),
-  parseSync: parseSHP
+  parseSync: parseSHP,
+  parseInBatches
 };
+
+async function* parseInBatches(asyncIterator, options) {
```

Comment:
Maybe add TODO that this should be handled by `parseInBatches`. In fact you might want to try your hand at making that change as well - in a separate PR of course.